diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8a92ad7..43b1e7d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,8 +25,16 @@ include_directories(${TANGO_PKG_INCLUDE_DIRS})
 link_directories(${TANGO_PKG_LIBRARY_DIRS})
 
 add_executable(Starter ${SOURCES} ${ADDITIONAL_SOURCES})
-target_link_libraries(Starter ${TANGO_PKG_LIBRARIES} -Wl,-z,now -pie)
-target_compile_options(Starter PUBLIC ${TANGO_PKG_CFLAGS_OTHER} -Wall -Wextra -D_FORTIFY_SOURCE=2 -O1 -fpie)
+target_link_libraries(Starter ${TANGO_PKG_LIBRARIES})
+target_compile_options(Starter PUBLIC ${TANGO_PKG_CFLAGS_OTHER} -Wall -Wextra -D_FORTIFY_SOURCE=2 -O1)
+
+message("Compiler: ${CMAKE_CXX_COMPILER_ID}")
+if(NOT (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" OR CMAKE_CXX_COMPILER_ID STREQUAL "Clang"))
+    target_compile_options(Starter PUBLIC -fpie)
+    set_target_properties(Starter PROPERTIES LINK_FLAGS "-Wl,-z,now,-pie")
+else()
+    set_target_properties(Starter PROPERTIES LINK_FLAGS "-bind_at_load")
+endif()
 
 install(TARGETS Starter
         RUNTIME DESTINATION "${CMAKE_INSTALL_FULL_BINDIR}"
