diff --git a/cmake/FindZeroMQ.cmake b/cmake/FindZeroMQ.cmake
index 4eb5e89..c772e04 100644
--- a/cmake/FindZeroMQ.cmake
+++ b/cmake/FindZeroMQ.cmake
@@ -66,14 +66,14 @@ find_path(ZeroMQ_INCLUDE_DIR
 
 if(WIN32)
     set(_zmq_versions "4_0_5" "4_3_4")
-    set(_zmq_vc_versions "v141" "v142")
+    set(_zmq_vc_versions "" "-v141" "-v142")
 
     foreach(ver IN LISTS _zmq_versions)
         foreach(vc_ver IN LISTS _zmq_vc_versions)
-            list(APPEND _zmq_lib_release_names "libzmq-${vc_ver}-mt-${ver}.lib")
-            list(APPEND _zmq_lib_debug_names "libzmq-${vc_ver}-mt-gd-${ver}.lib")
-            list(APPEND _zmq_lib_static_release_names "libzmq-${vc_ver}-mt-s-${ver}.lib")
-            list(APPEND _zmq_lib_static_debug_names "libzmq-${vc_ver}-mt-sgd-${ver}.lib")
+            list(APPEND _zmq_lib_release_names "libzmq${vc_ver}-mt-${ver}.lib")
+            list(APPEND _zmq_lib_debug_names "libzmq${vc_ver}-mt-gd-${ver}.lib")
+            list(APPEND _zmq_lib_static_release_names "libzmq${vc_ver}-mt-s-${ver}.lib")
+            list(APPEND _zmq_lib_static_debug_names "libzmq${vc_ver}-mt-sgd-${ver}.lib")
         endforeach(vc_ver IN LISTS VC_VERSIONS)
     endforeach(ver IN ZMQ_VERSIONS)
 
